tests/check/check_test.go

```diff
@@ -0,0 +1,133 @@
+package check
+
+import (
+	"context"
+	"log"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+	parser "github.com/craigpastro/openfga-dsl-parser/v2"
+	"github.com/openfga/openfga/cmd"
+	"github.com/openfga/openfga/pkg/testfixtures/storage"
+	"github.com/stretchr/testify/require"
+	pb "go.buf.build/openfga/go/openfga/api/openfga/v1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"gopkg.in/yaml.v2"
+)
+
+type checkTests struct {
+	Tests []checkTest
+}
+
+type checkTest struct {
+	Name   string
+	Stages []stage
+}
+
+type stage struct {
+	Model      string
+	Tuples     []*pb.TupleKey
+	Assertions []assertion
+}
+
+type assertion struct {
+	Tuple       *pb.TupleKey
+	Expectation bool
+}
+
+func TestCheckMemory(t *testing.T) {
+	testCheck(t, "memory")
+}
+
+func TestCheckPostgres(t *testing.T) {
+	testCheck(t, "postgres")
+}
+
+func TestCheckMySQL(t *testing.T) {
+	testCheck(t, "mysql")
+}
+
+func testCheck(t *testing.T, engine string) {
+	data, err := os.ReadFile("tests.yaml")
+	require.NoError(t, err)
+
+	var tests checkTests
+	err = yaml.Unmarshal(data, &tests)
+	require.NoError(t, err)
+
+	container := storage.RunDatastoreTestContainer(t, engine)
+
+	cfg := cmd.MustDefaultConfigWithRandomPorts()
+	cfg.Datastore.Engine = engine
+	cfg.Datastore.URI = container.GetConnectionURI()
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	go func() {
+		if err := cmd.RunServer(ctx, cfg); err != nil {
+			log.Fatal(err)
+		}
+	}()
+
+	conn, err := grpc.Dial(cfg.GRPC.Addr, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	require.NoError(t, err)
+	defer conn.Close()
+
+	client := pb.NewOpenFGAServiceClient(conn)
+
+	// Ensure the service is up before continuing.
+	policy := backoff.NewExponentialBackOff()
+	policy.MaxElapsedTime = 10 * time.Second
+	err = backoff.Retry(func() error {
+		_, err := client.CreateStore(ctx, &pb.CreateStoreRequest{Name: engine})
+		return err
+	}, policy)
+	require.NoError(t, err)
+
+	runTests(t, client, tests)
+
+	// Shutdown the server.
+	cancel()
+}
+
+func runTests(t *testing.T, client pb.OpenFGAServiceClient, tests checkTests) {
+	ctx := context.Background()
+
+	for _, test := range tests.Tests {
+		for _, stage := range test.Stages {
+			t.Run(test.Name, func(t *testing.T) {
+				resp, err := client.CreateStore(ctx, &pb.CreateStoreRequest{Name: "EnsureServiceIsUp"})
```

Comment:
This is a duplicate of the retry function above on lines 83-87.

Let's either a) use the Health Check endpoint for the "service up" check above or b) remove this code since we have a store created from our "service up" check above.

Reply:
Don't we want to run each test in a new store?

But, yes, the name was leftover from a refactor, so I updated that to use `test.Name`.

Reply:
Oh, right. Is it cool if we use the Health check endpoint for the readiness check instead then?

Like we're doing here https://github.com/openfga/openfga/blob/main/cmd/openfga/main_test.go#L162

For some reason this threw me off because of that.